### DISCUSSION TITLE
Detect usage of dynamic imports and put warning in console

### DISCUSSION THREAD

**Discussion by retrodaredevil** [grafana/grafana#47808] [Platform]
Hi, I am creating a plugin that has dependencies that use dynamic imports. I had no idea these dependencies had dynamic imports and I spent 2 hours trying to figure out why I was getting a bunch of `ChunkLoadErrors`. The issue here: https://github.com/grafana/grafana/issues/40671 describes a workaround to the problem, but I think there needs to be a better way to alert developers of this issue.

Yes, there is a section on the readme page about dynamic imports, but I had no idea that dynamic imports were even being used. Now that I know what the problem was, a fix doesn't benefit me, but I'm sure it will benefit other developers that run into this problem and get a bunch of 404 errors with no useful information to debug them.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
